docs/reference/how-to/recipes.asciidoc

```diff
@@ -223,3 +223,81 @@ In that case, since `ski` was in-between quotes, it was searched on the
 `1` matched. This allows users to mix exact search with stemmed search as they
 like.
 
+[float]
+[[consistent-scoring]]
+=== Getting consistent scoring
+
+The fact that Elasticsearch operates with shards and replicas adds challenges
+when it comes to having good scoring.
+
+[float]
+==== Scores are not reproducible
+
+Say the same user runs the same request twice in a row and documents do not come
+back in the same order both times, this is a pretty bad experience isn't it?
+Unfortunately this is something that can happen if you have replicas
+(`index.number_of_replicas` is greater than 0). The reason is that Elasticsearch
+selects the shards that the query should go to in a round-robin fashion, so it
+is quite likely if you run the same query twice in a row that it will go to
+different copies of the same shard.
+
+Now why is it a problem? Index statistics are an important part of the score.
+And these index statistics may be different across copies of the same shard
+due to deleted documents. As you may know when documents are deleted or updated,
+the old document is not immediately removed from the index, it is just marked
+as deleted and it will only be removed from disk on the next time that the
+segment this old document belongs to is merged. However for practical reasons,
```

Comment:
I'm not a fan of that `,`. There are lots of different sets of rules but I follow rules kind of like [these]( https://owl.english.purdue.edu/owl/resource/607/03/)